iree/compiler/Codegen/Common/SetNumWorkgroupsFromLinalgExtPass.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Dialect/LoweringConfig.h"
+#include "iree/compiler/Codegen/PassDetail.h"
+#include "iree/compiler/Codegen/Passes.h"
+#include "iree/compiler/Codegen/Transforms/Transforms.h"
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "iree/compiler/Dialect/HAL/IR/HALDialect.h"
+#include "iree/compiler/Dialect/HAL/IR/HALOps.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-codegen-set-num-workgroups-from-lianlg-ext"
+
+static const unsigned kNumMaxParallelDims = 3;
+
+static constexpr auto kFakeHALEntryPointRegionOpName =
```

Comment:
It would not be too much overhead to add an operation in `iree-dialects` that is `workgroup_id`/`workgroup_count`.

Reply:
Yes, these ones were easy (added in the parent commit).
The `entry_point` op was more annoying and I had to strip it down to nothing for now.